wire-ios-data-model/Source/MLS/Migration/ProteusToMLSMigrationCoordinator.swift

```diff
@@ -148,6 +147,28 @@ public class ProteusToMLSMigrationCoordinator: ProteusToMLSMigrationCoordinating
         return .canStart
     }
 
+    /// The migrateOrJoinGroupConversations method is responsible for processing a collection of team group conversations within a given context.
+    /// It evaluates each conversation to determine if it needs to be migrated or if it should join a group.
+    /// This method is crucial for maintaining the synchronization of group conversations,
+    ///  in scenarios where conversations may need to be migrated or newly integrated into groups.
+    ///
+    func migrateOrJoinGroupConversations() {
+        let conversations = try? ZMConversation.fetchAllTeamGroupConversations(messageProtocol: .mixed, in: context)
+        conversations?.forEach { conversation in
+            guard let groupID = conversation.mlsGroupID else { return }
+            guard let mlsService = context.mlsService else { return }
+            if mlsService.conversationExists(groupID: groupID) {
+                // if conversation exists we finalize migration
+
+            } else {
+                Task {
+                    try? await mlsService.joinGroup(with: groupID)
+                }
+            }
+        }
+
+    }
```

Comment:
Let's avoid optional `try` and catch and log errors instead. It will be helpful during manual testing to understand if anything is going wrong

Reply:
if possible make the full method async, conversationExists will be async too.

Reply:
conversationExists will become async on a different PR.